src/index.js

```diff
@@ -56,37 +64,46 @@ const requestFactory = (meth, notify = null) => {
       finalHeaders['Authorization'] = 'Bearer ' + token
     }
 
+    if (!('rtype' in options)){
+      options['rtype'] = 'json'
+    } else if (['json', 'file', 'string'].indexOf(options['rtype']) < 0){
+      Promise.reject(new BadReturnType(options['rtype']))
```

Comment:
I'm guessing this should return the Promise?

Reply:
I wasn't exactly sure what to do here. I want to throw an exception but from a quick google throwing an exception in a function that returns a promise isn't great (means you have to have a try catch block as well as a catch for the promise failing). Having it reject like this just means that instead of needing that try catch block, it gets handled by the catch for the promise failing. Thought it was a bit cleaner to deal with?

Reply:
If the function is always supposed to return a promise, you should return this rejected promise and handle the error in a catch of the function that calls this function.

Just doing `Promise.reject()` by itself won't do anything as far as I know.

Reply:
Try to name things as obviously as possible so a random developer could look at the code and have a good idea of what it does. So I would rename the array to something more like "availReturnTypes" or "allowedReturnTypes".

I find favouring clarity at the expense of conciseness helps with long-term maintainability.

Reply:
Yeah fair enough. And that promise.reject definitely should have been returned (it is in the newer commits)